### PR TITLE
Add pg_dump diff lock file generation

### PR DESCRIPTION
**Pull request by jld3103**

### CODE REVIEW

File: cmd/generate.go [hunk 1]
```diff
@@ -98,30 +90,212 @@ var generateCmd = &cobra.Command{
 					}
 				}
 			}
-		})
+		}()
 
-		run(config, newMigrationFilePath, migrationNumber)
+		err = run(config, newMigrationFilePath, migrationNumber)
+		if err != nil {
+			log.Fatalf("Failed to run: %v\n", err)
```
[the remark this answers is not in the document]
Comment (GeertJohan) [reply cmd/generate.go:97]: But maybe we shouldn't do above cleanup when the program fails?

File: cmd/generate.go [hunk 2]
```diff
@@ -98,30 +90,212 @@ var generateCmd = &cobra.Command{
...
+	return function(targetContainerID, migrateContainerID)
+}
+
+//nolint:gocognit
+func run(
+	config *internal.Config,
+	newMigrationFilePath string,
+	migrationNumber uint,
+) error {
+	return withPostgresContainers(func(targetContainerID, migrateContainerID string) error {
```
Comment (GeertJohan) [inline cmd/generate.go:138]: The `withPostgresContainers` and `PostgresFunction` setup feels a bit too complex honestly. It's only being used once, within run().

I think it's safe to remove the wrapping and just move the content of `withPostgresContainers` into `run`?

**Review by GeertJohan — CHANGES_REQUESTED** [review]

File: cmd/generate.go (hunk 2 above)
Comment (jld3103) [reply cmd/generate.go:138]: Yes, let's remove the wrapping. When I refactored it, it was more complex and it made sense to have this wrapping, but then I simplified it and forgot it.

File: cmd/generate.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jld3103) [reply cmd/generate.go:97]: Yes, I don't think we should run the cleanup function when it fails.

**Pull request merged**